packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -86,9 +86,59 @@ class _RunMetrics {
   final int childCount;
 }
 
+/// How the child is inscribed into the available space.
+///
+/// See also:
+///
+///  * [RenderWrap], the wrap render object.
+///  * [Wrap], the wrap widget.
+///  * [Wrapped], the widget to set the [WrapFit].
+enum WrapFit {
+  /// The child is placed either in the current or the next run, depending on
+  /// its min intrinsic size in the [Wrap.direction].
+  /// Within this run, it is forced to fill the entire run, unless the [Wrap]
+  /// has no max size constraint in the run direction.
+  ///
+  /// This setting is more expensive, because it also computes the minimal
+  /// size of the child. Avoid using it for complex children.
+  ///
+  /// The [Wrapped] widget assigns this kind of [WrapFit] to its child.
+  runTight(true),
```

Comment:
I've added `runMaybeTight` now.

I agree that the documentation is very important here. And, to be honest, I struggle to write a good documentation here.
Maybe it'd be best to also document these values visually using images and/or gifs like [this one](https://github.com/fischerscode/pub.packages/assets/45403027/f1bb12bb-ae49-42fc-a075-10c7be46c9f5).

From a first look, I like your wording better. But it might be best to ask someone who is unaware of the code, once we are happy with one.

Reply:
Alternative names:
- `loose` -> `standard` (default is a keyword)
- `runLoose` -> `constrained` (Constrained to a fitting run)
- `runTight` -> `fillRun` (Fill a fitting run)
- `tight` -> `fillNext` (Fill the next run)
- `runMaybeTight` -> `mightFill` (might fill the current run)

Reply:
I've changed the docs to your wording. (61884f94c5)

Reply:
I think I'll also have to write an article about `WrapFit`, the different options and how it can be used once this is merged.

Reply:
@nate-thegrate What do you think about [the alternative names](https://github.com/flutter/flutter/pull/136161#discussion_r1703400788)?

Reply:
Done (f32f0df7fb37bf7be36f2120a482ef55f12cb7aa)